File: gen_scripts/python3/scripts/f00.loop.py
#!/usr/bin/env python3
# testbox: requires=posix
# testbox: title="subshell (ls) with for..in collection"
import os

# collection loop on buffered output from subshell
for item in os.popen("ls dirtest").readlines():
  # test if item is directory
  path = "dirtest/" + item
  if os.path.isdir(path.rstrip()):
    print("%s is a directory" % item.rstrip())
  else:
    print("%s is not a directory" % item.rstrip())

File: lessons/gen_scripts/perl/scripts/f00.loop.pl
#!/usr/bin/env -S perl -w
# testbox: requires=posix
# testbox: title="subshell (ls) with foreach collection"

# collection loop on list returned by subshell
foreach $item (split /\s+/, `ls dirtest`) {
  if (-d "dirtest/$item") {
    print "$item is a directory\n"
  } else {
    print "$item is not a directory\n"
  }
}
